src/phoenix/server/api/types/node.py

```diff
@@ -1,36 +1,19 @@
-import base64
-import dataclasses
-from typing import Tuple, Union
+from typing import Tuple
 
-import strawberry
-from graphql import GraphQLID
-from strawberry.custom_scalar import ScalarDefinition
-from strawberry.schema.types.scalar import DEFAULT_SCALAR_REGISTRY
+from strawberry.relay import GlobalID
 
 
-def to_global_id(type_name: str, node_id: int) -> str:
-    """
-    Encode the given id into a global id.
-
-    :param type_name: The type of the node.
-    :param node_id: The id of the node.
-    :return: A global id.
-    """
-    return base64.b64encode(f"{type_name}:{node_id}".encode("utf-8")).decode()
-
-
-def from_global_id(global_id: str) -> Tuple[str, int]:
+def from_global_id(global_id: GlobalID) -> Tuple[str, int]:
     """
     Decode the given global id into a type and id.
 
     :param global_id: The global id to decode.
     :return: A tuple of type and id.
     """
-    type_name, node_id = base64.b64decode(global_id).decode().split(":")
-    return type_name, int(node_id)
+    return global_id.type_name, int(global_id.node_id)
 
 
-def from_global_id_with_expected_type(global_id: str, expected_type_name: str) -> int:
+def from_global_id_with_expected_type(global_id: GlobalID, expected_type_name: str) -> int:
```

Comment:
super-nit: we can probably cut these two down to just one function, since `expected_type_name` is sufficiently suggestive of the intent.
```suggestion
def from_global_id(global_id: GlobalID, expected_type_name: Optional[str] = None) -> int:
```

Reply:
The tricky part is that the return types are different on the two functions. I guess we could just always return the type name, which we don't currently do with `from_global_id_with_expected_type`? Going to punt for now.